### PR TITLE
Improve pad functionality, add tests and docs

### PR DESCRIPTION
**Pull request by pineapplemachine**

### CODE REVIEW

File: src/core/length.js [hunk 1]
```diff
@@ -1,7 +1,12 @@
...
+import {isSequence} from "./sequence";
+import {isArray, isFunction, isString} from "./types";
 
 export const canGetLength = (source) => {
-    return isString(source) || "length" in source;
+    return (
+        isString(source) || isArray(source) || (
+            isSequence(source) && source.length
+        )
+    );
```
Comment (pineapplemachine) [inline src/core/length.js:9]: Bugfix that was causing `startsWith` and `endsWith` to behave incorrectly

File: src/functions/pad.js [hunk 2]
```diff
@@ -1,251 +0,0 @@
...
-    },
-    rightCount: function(count, element){
-        if(this.source.unbounded()) return source;
-        else return count <= 0 ? source : new PadRightSequence(
-            this.source, element, count
-        );
-    },
-});
-
-export const pad = wrap({
```
Comment (pineapplemachine) [inline src/functions/pad.js:235]: Old pattern was mostly because `unordered` argument validation didn't exist yet

**Comment by pineapplemachine** [comment]
The difference between `padLeft`/`padRight` and `padLeftCount`/`padRightCount` is that the former pad an input sequence so that it's at least as long as some target length while the latter pad an input sequence with a given definite number of elements

File: src/functions/padRightCount.js [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+    name: "padRightCount",
+    summary: "Pad the front of an input sequence with some repeated element.",
+    docs: process.env.NODE_ENV !== "development" ? undefined : {
+        introduced: "higher@1.0.0",
+        examples: [
+            "basicUsage",
+        ],
+        related: [
+            "padLeft", "padRightCount",
+        ],
```
Comment (pineapplemachine) [inline src/functions/padRightCount.js:243]: TODO: Proper documentation for all of these functions

**Review by jpsullivan — APPROVED** [review]
So initially I had concerns regarding the `padXCount` method names, but after thinking on it for a couple days, I really can't think of anything else to call it. Makes perfect sense once you look at the tests, but it wasn't immediately obvious to me. Nonetheless, everything appears to be pretty sound here.

**Pull request merged**